File: src/com/silverkeytech/android_rivers/outlines/Outline.java
package com.silverkeytech.android_rivers.outlines;


import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.ElementList;

import java.util.ArrayList;
import java.util.List;

public class Outline {
    @Attribute(required = true)
    public String text;
    @Attribute(name = "type", required = false)
    public String outlineType;
    @Attribute(required = false)
    public String name;
    @Attribute(required = false)
    public String url;
    @Attribute(required = false)
    public String opmlUrl;
    @Attribute(required = false)
    public String htmlUrl;
    @Attribute(required = false)
    public String xmlUrl;
    @Attribute(required = false)
    public String language;

    @ElementList(inline = true, required = false)
    public List<Outline> outline = new ArrayList<Outline>();

    @Override
    public String toString() {
        return text;
    }
}

File: src/com/silverkeytech/android_rivers/outlines/Outline.kt
package com.silverkeytech.android_rivers.outlines

import java.util.jar.Attributes
import org.simpleframework.xml.Attribute

public class Outline{
    [Attribute] public var text : String? = null
    [Attribute(name = "type")] public var outlineType : String? = null
    [Attribute] public var name : String? = null
    [Attribute] public var url : String? = null
    [Attribute] public var opmlUrl : String? = null
    [Attribute] public var htmlUrl : String? = null
    [Attribute] public var xmlUrl : String? = null
    [Attribute] public var language : String? = null
}